src/components/schduler/weeklyView/table/table.tsx

```diff
@@ -0,0 +1,186 @@
+import React, { useState } from "react";
+import { UiElementProps } from "../../../common";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import {
+  faChevronLeft,
+  faChevronRight,
+} from "@fortawesome/free-solid-svg-icons";
+
+interface Event {
+  name: string;
+  slot?: any;
+  startTime: string;
+  endTime: string;
+  merge: boolean;
+}
+
+interface SchedulerTableProps extends UiElementProps {
+  events: Event[];
+}
+
+const getWeekStart = (date: Date): Date => {
```

Comment:
You can use the Luxon function "startOf" to get the start of a week

For example:

const today = DateTime.now();
const startOfWeek = today.startOf('week');